packages/camera/camera/ios/Classes/CameraPlugin.m

```diff
@@ -1115,11 +1118,16 @@ - (void)startImageStreamWithMessenger:(NSObject<FlutterBinaryMessenger> *)messen
     FlutterEventChannel *eventChannel =
         [FlutterEventChannel eventChannelWithName:@"plugins.flutter.io/camera/imageStream"
                                   binaryMessenger:messenger];
+    FLTThreadSafeEventChannel *threadSafeEventChannel =
+        [[FLTThreadSafeEventChannel alloc] initWithEventChannel:eventChannel];
 
     _imageStreamHandler = [[FLTImageStreamHandler alloc] init];
-    [eventChannel setStreamHandler:_imageStreamHandler];
-
-    _isStreamingImages = YES;
+    [threadSafeEventChannel setStreamHandler:_imageStreamHandler
```

Comment:
I missed this until it came up in a discussion with @gaaclarke : You're setting the stream handler safely now, but what about the calls to the event sink that's set up in the handler's callback? Are those already on the main thread?

Reply:
Very good callout! I missed the event sink calls in my original proposal (Just updated it and tagged you).